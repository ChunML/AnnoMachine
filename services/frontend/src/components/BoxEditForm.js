import React from 'react';
import PropTypes from 'prop-types';

function BoxEditForm({ coords, onInputChange }) {
  const { x_min, y_min, x_max, y_max } = coords;

  return (
    <div className="row space-around">
      <div className="input-field one-forth">
        <input
          type="text"
          name="x_min"
          value={x_min}
          onChange={onInputChange}
        />
      </div>
      <div className="input-field one-forth">
        <input
          type="text"
          name="y_min"
          value={y_min}
          onChange={onInputChange}
        />
      </div>
      <div className="input-field one-forth">
        <input
          type="text"
          name="x_max"
          value={x_max}
          onChange={onInputChange}
        />
      </div>
      <div className="input-field one-forth">
        <input
          type="text"
          name="y_max"
          value={y_max}
          onChange={onInputChange}
        />
      </div>
    </div>
  );
}

BoxEditForm.propTypes = {
  coords: PropTypes.object.isRequired,
  onInputChange: PropTypes.func.isRequired,
};

export default BoxEditForm;
