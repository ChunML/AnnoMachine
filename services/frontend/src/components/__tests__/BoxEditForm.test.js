import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import BoxEditForm from '../BoxEditForm';
import '../../setupTests';

const coords = {
  x_min: 10,
  y_min: 10,
  x_max: 20,
  y_max: 20,
};

const onInputChange = jest.fn();

it('BoxEditForm renders properly', () => {
  const wrapper = shallow(
    <BoxEditForm coords={coords} onInputChange={onInputChange} />
  );
  const form = wrapper.find('.row');
  expect(form.length).toBe(1);
  const inputs = form.find('input');
  expect(inputs.length).toBe(4);
  expect(inputs.get(0).props.name).toEqual('x_min');
  expect(inputs.get(1).props.name).toEqual('y_min');
  expect(inputs.get(2).props.name).toEqual('x_max');
  expect(inputs.get(3).props.name).toEqual('y_max');
  expect(onInputChange).toHaveBeenCalledTimes(0);
  inputs.at(0).simulate('change', { target: { name: 'x_min', value: '200' } });
  expect(onInputChange).toHaveBeenCalledTimes(1);
  inputs.at(0).simulate('change', { target: { name: 'y_min', value: '200' } });
  expect(onInputChange).toHaveBeenCalledTimes(2);
  inputs.at(0).simulate('change', { target: { name: 'x_max', value: '200' } });
  expect(onInputChange).toHaveBeenCalledTimes(3);
  inputs.at(0).simulate('change', { target: { name: 'y_max', value: '200' } });
  expect(onInputChange).toHaveBeenCalledTimes(4);
});

it('BoxEditForm renders a snapshot properly', () => {
  const tree = renderer
    .create(<BoxEditForm coords={coords} onInputChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
