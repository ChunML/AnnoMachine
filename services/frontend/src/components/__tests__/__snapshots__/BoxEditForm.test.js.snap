// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxEditForm renders a snapshot properly 1`] = `
<div
  className="ui form"
>
  <div
    className="fields"
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <div
      className="two wide field"
    >
      <input
        name="x_min"
        onChange={[MockFunction]}
        style={
          Object {
            "border": "none",
            "borderBottom": "1px solid #38ACEC",
            "borderRadius": "0",
            "color": "#38ACEC",
            "outline": "none",
          }
        }
        type="text"
        value={10}
      />
    </div>
    <div
      className="two wide field"
    >
      <input
        name="y_min"
        onChange={[MockFunction]}
        style={
          Object {
            "border": "none",
            "borderBottom": "1px solid #38ACEC",
            "borderRadius": "0",
            "color": "#38ACEC",
            "outline": "none",
          }
        }
        type="text"
        value={10}
      />
    </div>
    <div
      className="two wide field"
    >
      <input
        name="x_max"
        onChange={[MockFunction]}
        style={
          Object {
            "border": "none",
            "borderBottom": "1px solid #38ACEC",
            "borderRadius": "0",
            "color": "#38ACEC",
            "outline": "none",
          }
        }
        type="text"
        value={20}
      />
    </div>
    <div
      className="two wide field"
    >
      <input
        name="y_max"
        onChange={[MockFunction]}
        style={
          Object {
            "border": "none",
            "borderBottom": "1px solid #38ACEC",
            "borderRadius": "0",
            "color": "#38ACEC",
            "outline": "none",
          }
        }
        type="text"
        value={20}
      />
    </div>
  </div>
</div>
`;
